src/Tizen.Network.Stc/Tizen.Network.Stc/StcManager.cs

```diff
@@ -0,0 +1,112 @@
+    public sealed class SafeStcHandle : SafeHandle
+    {
+        internal SafeStcHandle() : base(IntPtr.Zero, true)
+        {
+        }
+
+        /// <summary>
+        /// Checks the validity of the handle.
+        /// </summary>
+        /// <value>Represents the validity of the handle.</value>
+        public override bool IsInvalid
+        {
+            get
+            {
+                return this.handle == IntPtr.Zero;
+            }
+        }
+
+        /// <summary>
+        /// Release the Stc handle
+        /// </summary>
+        /// <returns>True if the handle is released successfully, otherwise false.</returns>
+        protected override bool ReleaseHandle()
+        {
+            Interop.Stc.Deinitialize(this.handle);
+            this.SetHandle(IntPtr.Zero);
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// A class which is used to manage Smart Traffic control (Stc).<br/>
+    /// </summary>
+    public static class StcManager
+    {
+        /// <summary>
+        /// Gets the Stc safe handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <returns>The instance of the SafeStcHandle.</returns>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        internal static SafeStcHandle GetStcHandle()
+        {
+            return StcManagerImpl.Instance.GetSafeHandle();
+        }
+
+        /// <summary>
+        /// Gets the statistics information of an application, which matched the StatisticsFilter, asynchronously. Application ID must be provided.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="appId"> Application's appId for which statistics information is required.</param>
+        /// <param name="filter"> The StatisticsFilter object.</param>
+        /// <returns>The Statistics information of the application that used network in between timestamps specified.</returns>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        /// <exception cref="ArgumentException">Thrown when the method is provided with invalid argument.</exception>
+        public static Task<NetworkStatistics> GetStatisticsAsync(string appId, StatisticsFilter filter)
```

Comment:
According to your comment, The statistical data are split to some pieces.
So, the return value should be represented as collection of Network Statistics
Isn't it?

Reply:
I shall clear it once more from code flow and from the core developer.

Reply:
Yes, you are right. We will get many callbacks for 'GetStatisticsAsync'. Inside callback we can continue to store the statistics in list, but we don't know when to set taskResult. That is we don't know when will we get the last callback.
This similar case was faced in 'stc_foreach_stats'(deprecated). Now this is replaced by 'stc_foreach_stats'(async) and 'stc_foreach_stats'(sync).

Reply:
According to Mr. Tak, we should only use 'GetAllStatisticsAsync' for every purpose. Because this API is newly designed to solve the issue of 'last callback reached or not'.
Related ACR: http://suprem.sec.samsung.net/jira/browse/ACR-1358

Now we can provide AppId in StatisticsFilter according to need.